mysqld_exporter.go

```diff
@@ -347,6 +353,75 @@ var (
 	}
 )
 
+// whitelist for connection/process states in SHOW PROCESSLIST
+// tokudb uses the state column for "Queried about _______ rows"
+var (
+	// TODO: might need some more keys for other MySQL versions
+	// see https://dev.mysql.com/doc/refman/5.7/en/general-thread-states.html
+
+	// custom states (not coming from the state table)
+	// idle: "Sleep"(command)
+	// other query: state unknown, but command says query
+	// locked: blocked due to a lock (aggregation of system lock, waiting for ... lock, table lock)
+	// other: everything that could not be mapped and is not related to a query (e.g. slave thread)
+	tableStateWhitelist = map[string]bool{
+		"after create":              true,
+		"altering table":            true,
+		"analyzing":                 true,
+		"checking permissions":      true,
+		"checking table":            true,
+		"cleaning up":               true,
+		"closing tables":            true,
+		"converting heap to myisam": true,
+		"copying to tmp table":      true,
+		"creating sort index":       true,
+		"creating table":            true,
+		"creating tmp table":        true,
+		"executing":                 true,
+		"execution of init_command": true,
+		"end":                     true,
+		"freeing items":           true,
+		"flushing tables":         true,
+		"fulltext initialization": true,
+		"idle":                      true,
+		"init":                      true,
+		"killed":                    true,
+		"locked":                    true,
+		"logging slow query":        true,
+		"login":                     true,
+		"manage keys":               true,
+		"opening tables":            true,
+		"optimizing":                true,
+		"preparing":                 true,
+		"reading from net":          true,
+		"removing duplicates":       true,
+		"removing tmp table":        true,
+		"reopen tables":             true,
+		"repair by sorting":         true,
+		"repair done":               true,
+		"repair with keycache":      true,
+		"rolling back":              true,
+		"searching rows for update": true,
+		"sending data":              true,
+		"sorting for group":         true,
+		"sorting for order":         true,
+		"sorting index":             true,
+		"sorting result":            true,
+		"statistics":                true,
+		"updating":                  true,
+		"waiting for tables":        true,
+		"waiting for table flush":   true,
+		"waiting on cond":           true,
+		"writing to net":            true,
+		"other query":               true,
+		"other":                     true,
+	}
+	processlistDesc = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, processlist, "connection_count"),
+		"The number of times each user/host/state combination occured.",
+		[]string{"user", "host", "state"}, nil)
```

Comment:
This could get extremely large, and in addition it's strongly discouraged to have time-series that appear and disappear. I'd suggest keeping it to state at most.